### ISSUE TITLE
Support for models other than "User" is missing, despite allusions to it in the README

### ISSUE THREAD

**Issue by brandonzylstra** [jereinhardt/feste#25]
The README says 

```
In the model that holds your users' data, include Feste::User.
```

but it appears that in practice, _you *must* use the `User` model._

Without using the User model, I get `undefined method `subscriptions' for #<User:0x00007fc61c792658>` although I _nowhere_ specified that the `User` model ought to be expected to have such a method.  I did not put `include Feste::User` in the `User` model, and _did_ put it in a different model.

**Comment by jereinhardt** [comment]
@brandondrew thank you for filing this issue.  Where in your application is this error being raised?  If you could post a stacktrace, that would be helpful.  I have a suspicion of what might be causing this, but I need a little more information to confirm it.

**Comment by jereinhardt** [comment]
I an app to test my initial suspicions, and I think I know what may be happening in your application.  When a user visits the subscriptions path and there is no token in the query params to identify them, the controller will try to identify the subscriber by using whatever method was given to `config.authenticate_with`.  For instance, if you set this config to `:devise`, it will use whatever method the `Devise` gem uses to identify the current user, and assume that user is the subscriber.  

You can set this config to a `proc` which is called in the controller to identify the subscriber through a custom method.  If the model that uses `Feste::Subscriber` is different from the model that is returned from your authentication library, then setting `config.authenticate_with` to a custom proc is probably what you need to do.

As an example, if your subscribing model is `Student` and you have a way of identifying the current student in the controller (let's say you have `session[:student_id]` set to their id), then you would want to use something like this:

```ruby
authentication_method = Proc.new do |controller|
  ::Student.find_by(id: controller.session[:student_id])
end

Feste.configure do |config|
  config.authenticate_with = authentication_method
end
```

See if using something like that helps

**Comment by brandonzylstra** [comment]
actually, my subscribing model is `Region`, so everything you wrote above _probably_ applies, with the one caveat that it's not clear to me how to best handle this, for example designate the administrative user who can control the subscriptions (since each region will have a group e-mail address).  The thing that makes it unclear to me whether Feste will fit or not is that it's not just that my users are named differently, but rather the issue is that my subscriptions are not _per user,_ and your example seems to suggest a per-user approach that just involves a user model named something other than `User`.  Or am I wrong?

If you have any ideas please let me know.  For the moment I built what I needed without using Feste, but I am hoping to use Feste as soon as the next set of regional e-mail reports need to be added, assuming it can be used.

**Comment by jereinhardt** [comment]
> but rather the issue is that my subscriptions are not per user, and your example seems to suggest a per-user approach that just involves a user model named something other than User

You are absolutely right.  The guiding principle behind Feste is that it gives users the ability to manage their own email subscriptions.  Because of this, it creates subscriptions on a per-user basis, and creates a panel where users can select what emails they do and do not receive.

Based on the details you provided, it sounds like your use case conflicts with Feste's guiding principal in two ways.  The first is that it sounds like you want an administrative user to be able to dictate what emails other users receive.  The second is that each subscription would affect a group of users, rather than an individual.

There are a few things you could do that might make Feste work for you, but I think they would require certain practices on your end that would ultimately lead to a pretty brittle solution.  In this case, rolling your own subscription system might be the way to go.

**Comment by brandonzylstra** [comment]
> The first is that it sounds like you want an administrative user to be able to dictate what emails other users receive. The second is that each subscription would affect a group of users, rather than an individual.

Yes, these are correct, although (in case this changes anything) there is still a one-to-one relationship between a subscription and an e-mail address.  The fact that a group receives the messages is not something that the app needs to be concerned with.  (For instance, imagine "Region 1" with "region1@example.com".  The e-mail system handles expanding that to "user1@example.com", "user2@example.com", etc.)

Based on the statements above, am I correct to understand that although Feste can be used with a model named something other than `User`, it must still be _a model that represents **the user?**_

**Comment by jereinhardt** [comment]
Yes, the model that is used with Feste should directly represent the recipient of the email.
